cmd/apichecker/apicheck_test.go

```diff
@@ -0,0 +1,692 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/common/convert"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+var (
+	testAPIKey      = ""
+	testAPIToken    = ""
+	testChecklistID = ""
+	testCardID      = ""
+	testListID      = ""
+	testBoardID     = ""
+	testBoardName   = ""
+	canTestMainFile = false
+)
+
+func TestMain(m *testing.M) {
+	setTestVars()
+	testMode = true
+	c := log.GenDefaultSettings()
+	c.Enabled = convert.BoolPtr(true)
+	log.GlobalLogConfig = &c
+	log.Infoln(log.Global, "set verbose to true for more detailed output")
+	var err error
+	configData, err = readFileData(jsonFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	testConfigData, err = readFileData(testJSONFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	usageData = testConfigData
+	setTestVars()
+	defer os.Exit(m.Run())
+	defer func() {
+		err := removeTestFileVars()
+		if err != nil {
+			log.Error(log.Global, err)
+			os.Exit(1)
+		}
+	}()
+}
+
+func setTestVars() {
+	if !canUpdateTrello() {
+		apiKey = testAPIKey
+		apiToken = testAPIToken
+		trelloChecklistID = testChecklistID
+		trelloCardID = testCardID
+		trelloListID = testListID
+		trelloBoardID = testBoardID
+		trelloBoardName = testBoardName
+		return
+	}
+	setAuthVars()
+	testAPIKey = apiKey
+	testAPIToken = apiToken
+	testChecklistID = trelloChecklistID
+	testCardID = trelloCardID
+	testListID = trelloListID
+	testBoardID = trelloBoardID
+	testBoardName = trelloBoardName
```

Comment:
I think you should remove this section of code. I don't think we should be using the keys from the main json file

Reply:
thats the section thats allowing application to not crash after running with --create n firstrun. coz when we run test and the canTestMainFile is false, it cant access the main file because the testing is all done thro the test file unless that variable is set true so it was saying invalid keys yesterday